controllers/glance_controller.go

```diff
@@ -886,7 +893,17 @@ func (r *GlanceReconciler) generateServiceConfig(
 ) error {
 	labels := labels.GetLabels(instance, labels.GetGroupLabel(glance.ServiceName), serviceLabels)
 
-	ospSecret, _, err := secret.GetSecret(ctx, h, instance.Spec.Secret, instance.Namespace)
+	// ensure a MariaDBAccount CR and Secret exist.
+	// this mariadb API function will as an interim step actually generate a
+	// new MariaDBAccount and Secret if one does not exist already.   in a
+	// future release, this function may change to emit an error if the
+	// MariaDBAccount was not already created ahead of time (e.g. by openstack-operator
+	// or end-user YAML declaration)
+	databaseAccount, dbSecret, err := mariadbv1.EnsureMariaDBAccount(
```

Comment:
+1 this looks better and we don't have to revisit this code if something changes in mariadb. Thanks

Reply:
note the RequireTLS flag (i.e. create the MariaDB username as requiring TLS to log in) hardcoded to false right now in this call.  that...has to come from somewhere at some point

Reply:
I think at some point who creates the `MariaDBAccount` CR in advance (`openstack-operator`?) should pass that value to the CR: if we decide to expose that parameter (in a follow up work) we should have a struct for Database where we collect all the elements we need to set in the CR (`databaseAccount: string`, `requireTLS: bool`) and so forth. Having a high level view in the form of a doc patch would help moving forward w/ these kind of conversations in a different context.

Reply:
I think `Database` would not have extra fields for this, because the requireTLS aspect is going to be transparent to the downstream controllers.  the `my.cnf` file that gets shuttled over will capture whats needed here for the login to proceed.

Reply:
also AFAICT the `Database` facade seems to be used only for sending instructions to create databases, it does not seem to be used for consumption of parameters, at least here in glance though id have to look at nova and others to see what they do.   so at the moment i noted username/password fields would go away here, see my comment at https://github.com/openstack-k8s-operators/mariadb-operator/pull/184/files#r1484411670